appcelerator/titanium_mobile
# refactor(android): replace SDK "build.properties" with "BuildConfig"

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27574

**Summary:**
- Changed how Titanium version info is stored in library.
- Fixed show-stopper issue caused by [TIMOB-27778](https://jira.appcelerator.org/browse/TIMOB-27778) that causes library to wrongly report version "1.0.0" for the 1st build. (This affects analytics too.)

**Test:**
1. Build and run the below on Android.
2. Verify alert's "Ti.version" matches SDK version. (ex: "9.1.0")
3. Verify alert's "Ti.buildHash" is **not** set to "HEAD".
4. Verify alert's "Ti.buildDate" is set to a recent date. (The time SDK was built by Jenkins.)
5. In the log, verify the "Powered by Titanium" version info looks correct.
Example: `[INFO]  MyApp 1.0 (Powered by Titanium 9.1.0.b6fc721175)`

```javascript
var message =
	"Ti.version: " + Ti.version + "\n" +
	"Ti.buildHash: " + Ti.buildHash + "\n" +
	"Ti.buildDate: " + Ti.buildDate;
alert(message);
```
